integration-tests/src/test/java/tachyon/master/JournalCrashIntegrationTest.java

```diff
@@ -0,0 +1,172 @@
+ * the License.
+ */
+
+package tachyon.master;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.client.TachyonFS;
+import tachyon.conf.TachyonConf;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.thrift.InvalidPathException;
+import tachyon.thrift.TableDoesNotExistException;
+import tachyon.util.CommonUtils;
+
+/**
+ * Test master journal for crashes. Assert that test can replay the editlog and reproduce the
+ * correct state. Test both the single master(tachyon) and multi masters(tachyon-ft) mode.
+ */
+public class JournalCrashIntegrationTest {
+
+  /**
+   * Hold a client and keep creating files/tables.
+   */
+  class ClientThread implements Runnable {
+    /** The number of successfully created files/tables. */
+    public int mSuccessNum = 0;
+
+    private final int mOpType;  //0:create file; 1:create raw table.
+    private final TachyonFS mTfs;
+
+    public ClientThread(int opType, TachyonFS tfs) {
+      mOpType = opType;
+      mTfs = tfs;
+    }
+
+    @Override
+    public void run() {
+      try {
+        while (true) {
+          if (mOpType == 0) {
+            if (mTfs.createFile(new TachyonURI(TEST_FILE_DIR + mSuccessNum)) == -1) {
+              break;
+            }
+          } else if (mOpType == 1) {
+            if (mTfs.createRawTable(new TachyonURI(TEST_TABLE_DIR + mSuccessNum), 1) == -1) {
+              break;
+            }
+          }
+          mSuccessNum ++;
+          CommonUtils.sleepMs(null, 100);
+        }
+      } catch (Exception e) {
+        // Something crashed. Stop the thread.
+      }
+    }
+  }
+
+  private static final int TEST_BLOCK_SIZE = 128;
+  private static final String TEST_FILE_DIR = "/files/";
+  private static final int TEST_MASTERS = 3;
+  private static final String TEST_TABLE_DIR = "/tables/";
+  private static final long TEST_TIME_MS = Constants.SECOND_MS;
+
+  private ClientThread mCreateFileThread = null;
+  private ClientThread mCreateTableThread = null;
+  private final ExecutorService mExecutorsForClient = Executors.newFixedThreadPool(2);
+  private final ExecutorService mExecutorsForMaster = Executors.newFixedThreadPool(2);
+  private TachyonConf mMasterTachyonConf =  null;
+
+  @After
+  public final void after() throws Exception {
+    mExecutorsForClient.shutdown();
+    mExecutorsForMaster.shutdown();
+    System.clearProperty("fs.hdfs.impl.disable.cache");
+  }
+
+  @Before
+  public final void before() throws Exception {
+    System.setProperty("fs.hdfs.impl.disable.cache", "true");
+  }
+
+  /**
+   * Check if the correct state can be reproduced.
+   */
+  private void crashTestUtil(int successFiles, int successTables) throws IOException,
```

Comment:
When I see "util" I think of a util class, not a method name. Since this is ultimately checking the results, it should have "check" or "verify" as part of the name.